catalogs/catalog-lakehouse-hudi/src/main/java/org/apache/gravitino/catalog/lakehouse/hudi/backend/hms/HudiHMSBackendOps.java

```diff
@@ -73,12 +126,57 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
 
   @Override
   public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaException {
-    throw new UnsupportedOperationException("Not implemented yet");
+    NameIdentifier schemaIdent = NameIdentifier.of(namespace.levels());
+    if (!schemaExists(schemaIdent)) {
+      throw new NoSuchSchemaException("Schema (database) does not exist %s", namespace);
+    }
+
+    try {
+      return clientPool.run(
+          c -> {
+            List<String> allTables = c.getAllTables(schemaIdent.name());
+            return c.getTableObjectsByName(schemaIdent.name(), allTables).stream()
+                .filter(
+                    t ->
+                        t.getSd().getInputFormat() != null
+                            && t.getSd().getInputFormat().startsWith("org.apache.hudi"))
```

Comment:
Can you please make this "org.apache.hudi" a static variable to avoid hard coding here? Also adding some comments about the purpose here. This is quite hacky, because if the Hudi package is changed or somehow, the assumption here will be failed.

Reply:
fixed